### PR TITLE
Add garden helper functions [STOMAIL-7555]

### PR DESCRIPTION
**Pull request by pavel-mailpoet**
## Description

_N/A_

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

[STOMAIL-7555]

## After-merge notes

_N/A_

## Tasks

- [ ] I have added a changelog entry (`./do changelog:add --type=<type> --description=<description>`)
- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:add-ciab-helper-functions)

_The latest successful build from `add-ciab-helper-functions` will be used. If none is available, the link won't work._

## Summary by CodeRabbit

* **New Features**
  * Improved Garden environment detection and API to check Garden status.
  * Expose a site’s Garden name and Garden partner when available to improve contextual display and integrations.

* **Tests**
  * Added unit tests covering Garden detection and retrieval of Garden name/partner, including edge cases and fallback scenarios.

### CODE REVIEW

File: mailpoet/lib/WPCOM/DotcomHelperFunctions.php [hunk 1]
```diff
@@ -48,6 +48,40 @@ public function isEcommerce(): bool {
...
+    return $site ? \is_blog_garden($site) : false;
+  }
+
+  protected function getSiteMetaValue(string $meta_key): ?string {
+    if (!function_exists('get_site_meta')) {
+      return null;
+    }
+    $blog_id = \get_current_blog_id();
+    $value = \get_site_meta($blog_id, $meta_key, true);
+    return is_string($value) && $value !== '' ? $value : null;
```
Comment (triple0t) [inline mailpoet/lib/WPCOM/DotcomHelperFunctions.php:68]:
```suggestion
    return empty($value) ?  null : $value;
```
Since the return type of [get_site_meta](https://developer.wordpress.org/reference/functions/get_site_meta/) can be mixed. Why not use empty?

**Review by triple0t — DISMISSED** [review]
Looks good to me.
I dropped a minor feedback.

Feel free to update or move forward.

Well done.

File: mailpoet/lib/WPCOM/DotcomHelperFunctions.php (hunk 1 above)
Comment (pavel-mailpoet) [reply mailpoet/lib/WPCOM/DotcomHelperFunctions.php:68]: Thank you @triple0t for your suggestion. It makes sense, and I used your version. Quick question: how did you do that? I mean, how did you suggest a change that I could simply commit here on GitHub?

Comment (pavel-mailpoet) [reply mailpoet/lib/WPCOM/DotcomHelperFunctions.php:68]: After the build failed, I know why I used my original version. It is because we don’t want to return `mixed` from this function; we want a string. So I am returning my original version.

Comment (triple0t) [reply mailpoet/lib/WPCOM/DotcomHelperFunctions.php:68]: > Quick question: how did you do that? I mean, how did you suggest a change that I could simply commit here on GitHub?

This is done with the "add suggestion" command
Screenshot 2025-09-09 at 11 45 06 AM

More info: https://egghead.io/lessons/github-add-suggestions-in-a-github-pr-review
